pkg/image/api/helper.go

```diff
@@ -903,3 +904,41 @@ func SetContainerImageEntrypointAnnotation(annotations map[string]string, contai
 func LabelForStream(stream *ImageStream) string {
 	return fmt.Sprintf("%s/%s", stream.Namespace, stream.Name)
 }
+
+// ImageHasSignature returns true if the signature identified by type and blob is already stored in the image.
+func ImageHasSignature(image *Image, sType string, blob []byte) bool {
+	for _, signature := range image.Signatures {
+		if signature.Type == sType && bytes.Equal(blob, signature.Content) {
+			return true
+		}
+	}
+	return false
+}
+
+// AddImageSignature appends given signature to the image if it doesn't already have it.
+func AddImageSignature(image *Image, sType string, blob []byte) {
+	if !ImageHasSignature(image, sType, blob) {
+		s := ImageSignature{
+			Type:    sType,
+			Content: blob,
+		}
+		image.Signatures = append(image.Signatures, s)
+	}
+}
+
+// RemoveImageSignature removes signature identified by type and blob from the image if it has it.
+func RemoveImageSignature(image *Image, sType string, blob []byte) {
```

Comment:
Make it an `IndexOfImageSignature` function, then use that value to do a `copy` operation to collapse the list in one go.